site/src/components/DeploySettingsLayout/Option.tsx

```diff
@@ -104,68 +110,65 @@ export const OptionValue: FC<OptionValueProps> = (props) => {
   return <span css={styles.option}>{JSON.stringify(value)}</span>;
 };
 
-interface OptionConfigProps extends HTMLAttributes<HTMLDivElement> {
-  source?: boolean;
-}
+type OptionConfigContextValue = {
+  isSource?: boolean;
+};
+
+const OptionConfigContext = createContext<OptionConfigContextValue>({
+  isSource: false,
+});
+
+type OptionConfigProps = HTMLAttributes<HTMLDivElement> &
+  OptionConfigContextValue;
 
-// OptionConfig takes a source bool to indicate if the Option is the source of the configured value.
-export const OptionConfig: FC<OptionConfigProps> = ({
-  children,
-  source,
-  ...attrs
-}) => {
+// OptionConfig takes a isSource bool to indicate if the Option is the source of the configured value.
+export const OptionConfig = ({ isSource, ...attrs }: OptionConfigProps) => {
   const theme = useTheme();
-  const borderColor = source ? undefined : theme.palette.divider;
+  const borderColor = isSource
+    ? theme.experimental.roles.active.outline
+    : theme.palette.divider;
 
   return (
-    <div
-      {...attrs}
-      css={{
-        fontSize: 13,
-        fontFamily: MONOSPACE_FONT_FAMILY,
-        fontWeight: 600,
-        backgroundColor: source
-          ? theme.palette.primary.dark
-          : theme.palette.background.paper,
-        display: "inline-flex",
-        alignItems: "center",
-        borderRadius: 2,
-        padding: "0 8px",
-        border: `1px solid ${borderColor}`,
-      }}
-    >
-      {children}
-    </div>
+    <OptionConfigContext.Provider value={{ isSource }}>
+      <div
+        {...attrs}
+        css={{
+          fontSize: 13,
+          fontFamily: MONOSPACE_FONT_FAMILY,
+          fontWeight: 600,
+          backgroundColor: theme.palette.background.paper,
+          display: "inline-flex",
+          alignItems: "center",
+          borderRadius: 4,
+          padding: 6,
+          lineHeight: 1,
+          gap: 6,
+          border: `1px solid ${borderColor}`,
+        }}
+      />
+    </OptionConfigContext.Provider>
   );
 };
 
-interface OptionConfigFlagProps extends HTMLAttributes<HTMLDivElement> {
-  source?: boolean;
-}
-
-export const OptionConfigFlag: FC<OptionConfigFlagProps> = ({
-  children,
-  source,
-  ...attrs
-}) => {
+export const OptionConfigFlag = (props: HTMLAttributes<HTMLDivElement>) => {
   const theme = useTheme();
+  const { isSource } = useContext(OptionConfigContext);
```

Comment:
Is there a way to target the `background-color` via CSS nesting in the parent? Introducing a new context for a style ternary in one place seems a little heavy-handed to me.

Reply:
I was wondering the same 🤔 going to make that instead.